### PR TITLE
[3.3.5] Core/Spells: Improved Devouring Plague spell direct damage

### LINKED ISSUE #14869: [3.3.5] Improved Devouring Plague error in direct damage
The Improved Devouring Plague talent is in error in the application of direct damage. He is totally ignoring the talent Shadow Weaving. Regardless of the number of stacks of Shadow Weaving the value of the direct damage of the Devouring Plague is not modified, only damage over time is modified.

Improved Devouring Plague Ids:

http://wotlk.openwow.com/spell=63625
http://wotlk.openwow.com/spell=63626
http://wotlk.openwow.com/spell=63627

Shadow Weaving:

http://wotlk.openwow.com/spell=15257
http://wotlk.openwow.com/spell=15331
http://wotlk.openwow.com/spell=15332

How to reproduce: Use Devouring Plague on a target.
Use Mind Fly to the Shadow Weaving accumulate 5 times. Use again Devouring Plague. Note that the Direct Damage has not undergone any modification.
Testes on rev. 3aa177410010e75934703be9247245881be73490

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/21556743-3-3-5-improved-devouring-plague-error-in-direct-damage?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
by @robinsch; closes #14869 (Improved Devouring Plague error in direct damage)

Pct damage bonus is not applied because in SpellDamageBonusDone,
it just returns 1.0f for DamageType DoT. Because Improved Devouring Plague direct damage
is based on DoT damage from plague, it should also be modified when DoT damage is.

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -1305,6 +1305,7 @@ void Aura::HandleAuraSpecificMods(AuraApplication const* aurApp, Unit* caster, b
                     if (AuraEffect const* aurEff = caster->GetDummyAuraEffect(SPELLFAMILY_PRIEST, 3790, 1))
                     {
                         uint32 damage = caster->SpellDamageBonusDone(target, GetSpellInfo(), GetEffect(0)->GetAmount(), DOT);
+                        damage *= caster->SpellDamagePctDone(target, GetSpellInfo(), SPELL_DIRECT_DAMAGE);
```
Comment (ShinDarth) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: don't we lose this after next assignment?

**Comment by Keader** [comment]
Running smoothly. :+1:

**Comment by tkrokli** [comment]
Yes, I suppose it should, seeing that it is only 1 line added to the Spell Damage Bonus calculation.
The only unverified question I can see is this: is the damage correctly calculated and blizzlike?

**Comment by robinsch** [comment]
Yes as stated in https://github.com/TrinityCore/TrinityCore/issues/14869

**Comment by tkrokli** [comment]
Very well. I suppose all that is left is to see when the devs want to commit this PR to the source.

**Comment by mik1893** [comment]
I'm merging this. For ALL other spell having this issue i will collect and make another commit. Please check #14918 and do test

**Comment by Keader** [comment]
Maybe it's also related to #14958

**Pull request merged**